solutions/cses-1640.mdx

```diff
@@ -118,6 +118,48 @@ int main() {
 ```
 
 </CPPSection>
+	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+public class SumOfTwoValues {
+	public static void main (String[] args) throws IOException {
+		Kattio io = new Kattio();
+
+		int n = io.nextInt();
+		int target = io.nextInt();
+		int[] values = new int[n];
+		// use a map to avoid using a very large array
+		Map <Integer, Integer> map = new HashMap<>();
```

Comment:
```suggestion
		Map<Integer, Integer> map = new HashMap<>();
```

Reply:
also put this above the last for loop

Reply:
```suggestion
		HashMap<Integer, Integer> map = new HashMap<>();
```
This is personal preference, but since the map isn't being set anywhere else, you might as well make it HashMap.

Reply:
He's doing it because of the polymorphism guideline: https://usaco.guide/general/adding-solution?lang=java#java